pkgs/by-name/ni/nix-init/package.nix

```diff
@@ -82,7 +82,7 @@ rustPlatform.buildRustPackage rec {
     GEN_ARTIFACTS = "artifacts";
     # FIXME: our libgit2 is currently too new
     # LIBGIT2_NO_VENDOR = 1;
-    NIX = lib.getExe nix;
+    NIX = lib.getExe nixForLinking;
```

Comment:
When looking at https://github.com/search?q=repo%3Anix-community%2Fnix-init+NIX+language%3ARust&type=code I don't think that is actually compiled in and if the env is not provided it just tried tries to execute "nix". Maybe we should move that to preCheck or so to make it clear.

Reply:
Ah, it is built in (`option_env!()` expands to the environment varible's value at compile time) but it's only using the `$NIX` variable for a command to execute, so we can keep this as `pkgs.nix`. Great catch, thanks!
https://doc.rust-lang.org/stable/std/macro.option_env.html